hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/ListBucketHandler.java

```diff
@@ -56,12 +61,15 @@ protected void execute(OzoneClient client, OzoneAddress address)
     int counter = 0;
     while (bucketIterator.hasNext() && counter < listOptions.getLimit()) {
       OzoneBucket bucket = bucketIterator.next();
-      if (bucket.isLink()) {
-        bucketList.add(new InfoBucketHandler.LinkBucket(bucket));
-      } else {
-        bucketList.add(bucket);
+      if (!encryptedBuckets ||
+              (encryptedBuckets && (bucket.getEncryptionKeyName() != null))) {
```

Comment:
Nit: condition can be simplified.  (Sorry for missing this earlier.)

```suggestion
      if (!encryptedBuckets || bucket.getEncryptionKeyName() != null) {
```